cudapoa/tests/Test_CudapoaNW.cu

```diff
@@ -304,6 +305,220 @@ TEST_P(NWTest, TestNWCorrectness)
 
 INSTANTIATE_TEST_SUITE_P(TestNW, NWTest, ValuesIn(getNWTestCases()));
 
+//---------------------------------------------------------------------------------------
+
+// host function for calling the kernels to test static/adaptive-band NW with/without traceback buffer
+NWAnswer testNWbanded(const BasicNW& obj, bool adaptive, bool traceback = false)
+{
+    typedef int16_t SizeT;
+
+    //declare device buffer
+    uint8_t* nodes;
+    SizeT* graph;
+    SizeT* node_id_to_pos;
+    SizeT graph_count; //local
+    uint16_t* incoming_edge_count;
+    SizeT* incoming_edges;
+    uint16_t* outgoing_edge_count;
+    SizeT* outgoing_edges;
+    uint8_t* read;
+    uint16_t read_count; //local
+    int16_t* scores;
+    int16_t* traces;
+    SizeT* alignment_graph;
+    SizeT* alignment_read;
+    int32_t gap_score;
+    int32_t mismatch_score;
+    int32_t match_score;
+    SizeT* aligned_nodes; //local; to store num of nodes aligned (length of alignment_graph and alignment_read)
```

Comment:
Initialize the pointers to `nullptr` to fail hard in case you forget to reset any of them. :boom: ;)

Reply:
Also initialize `graph_count` and `read_count` to a sensible value? Probably 0?

Reply:
done.